### PR TITLE
Terraform v0.13.x support

### PR DESCRIPTION
**Pull request by ianstanton**
- Bump aurora module for terraform v0.13 support
- Bump eks module for terraform v0.13 support
- Update pipeline for terraform v0.13 support
- Add required providers to this module

Related to https://github.com/astronomer/issues/issues/1910

### CODE REVIEW

**Comment by sjmiller609** [comment]
Adding @tommyk to review because he has been looking at google tf, it's probably worth getting him plugged in.

**Review by sjmiller609 — APPROVED** [review]

**Comment by sjmiller609** [comment]
@tommyk this module is for deploying the AWS infrastructure for Astronomer. The module/terraform-aws-astronomer-enterprise combines this module with the "system-components" and "kubernetes-astronomer" modules, both of which are shared with the google terraform. The original concept was to make the system components and app layers modular to plug in on top of the base terraform modules, but in hindsight I would instead keep them decoupled and remove application provisioning logic from the terraform. Instead, Astronomer could provide terraform for the base infrastructure, then ask users to install using Helm on top of that. In so doing, users could use the TF for ongoing maintenance of the infra and use helm + astronomer tooling for maintenance of the application. Terraform has not been well suited for Astronomer application day 2.

File: versions.tf [hunk 1]
```diff
@@ -1,4 +1,33 @@
...
+  required_version = ">= 0.13"
+  required_providers {
+    acme = {
+      source = "terraform-providers/acme"
+    }
+    archive = {
+      source = "hashicorp/archive"
+    }
+    aws = {
+      source = "hashicorp/aws"
```
Comment (sjmiller609) [inline versions.tf]: I think the change is good, but I wonder if we can version pin to the major or minor version some providers, in particular the AWS provider. WDYT @ianstanton?

File: eks.tf [hunk 2]
```diff
@@ -27,7 +27,7 @@ module "eks" {
   # copy of the pending PRs
   source = "terraform-aws-modules/eks/aws"
   # version of the eks module to use
-  version = "8.1.0"
+  version = "13.2.1"
```
Comment (danielhoherd) [inline eks.tf:30]: Make sure to run `terraform fmt` on this repo during the upgrade. There is already a `pre-commit` hook to do this, so make sure you've run `pre-commit install` in the git root to make use of that.

**Comment by ianstanton** [comment]
Converting to draft. I think there is an issue with `cluster-autoscaler` and the eks module version we're bumping to here.

**Comment by ianstanton** [comment]
The cluster-autoscaler issue is resolved after adding the worker-autoscaling policy. This was removed from the EKS module, so we lost it when updating that module.

File: versions.tf (hunk 1 above)
Comment (ianstanton) [reply versions.tf]: I don't see an issue with that. This was generated by `terraform`, so I'll add those and see if it causes any unexpected issues.

**Comment by ianstanton** [comment]
Draft until CI is moved to Circle.

**Comment by ianstanton** [comment]
Closing in favor of https://github.com/astronomer/terraform-aws-astronomer-aws/pull/54

**Pull request closed without merging**